FirmataParser.cpp

```diff
@@ -130,6 +130,9 @@ void FirmataParser::parse(uint8_t inputData)
           if (currentReportDigitalCallback)
             (*currentReportDigitalCallback)(currentReportDigitalCallbackContext, multiByteChannel, dataBuffer[0]);
           break;
+        case REPORT_VERSION:
+          if (currentReportVersionCallback)
+            (*currentReportVersionCallback)(currentReportVersionCallbackContext, dataBuffer[0], dataBuffer[1], (const char *)NULL);
```

Comment:
Originally, the parser was built into FirmataClass which serves the host/server.
There was no notion of a callback for version, ever; it was tightly coupled.
`REPORT_VERSION` was used to query the version from the host/server, and it would subsequently report the version using `REPORT_VERSION`.
The fact that `REPORT_VERSION` is a single byte operation in the protocol documentation is wrong, because when the version is actually reported it sends three bytes.
Therefore, `REPORT_VERSION` should always send three bytes and the server/host should be smart enough realize it is being queried and discard them.
An alternate approach would be to create a `QUERY_VERSION` single byte used to explicitly query the version from the server/host